crates/node/src/scheduler/mod.rs

```diff
@@ -240,6 +240,27 @@ impl Scheduler {
                 }
             };
 
+            //copy Task file to VM workspace
+            //Validate that everything is ok before starting the task.
+            for file in task.files.iter() {
+                if let Err(err) = file
+                    .copy_file_for_vm_exe(&self.data_directory, task.tx)
+                    .await
+                {
+                    tracing::error!(
+                        "An error occurs during Tx:{} task file copy for VM execution {err}",
+                        task.tx
+                    );
+                    if let Err(err) = self.database.mark_tx_executed(&task.tx).await {
+                        tracing::error!(
+                            "failed to update transaction.executed => true - tx.hash: {}",
+                            task.tx
+                        );
+                    }
```

Comment:
I'm a bit unsure about this. If task execution in VM fails, we don't mark the tx as executed and that has pros and cons. So far my thinking has been that it's good as it can be retried (after the technical error has been fixed).

This one is similar: One obvious case is when disk is full. The transaction could be executed when the disk has been cleaned.

On the other hand there can be a corner case where transaction could get stuck forever, which is not ideal either, but these individual exceptions can be manually fixed from DB.

WDYT?

Reply:
I agree, Tx re-execution is not easy to define because it depends on the cause. In our implementation the logic is simple, the node tries to re-execute a little later.  It leads to Tx that re-execute all the time until the reason is solved or the node restarted.
I think we can consider that if the Tx fail on one node, there's 2 case:
 * it fails only on that node (disk full for example), the user can get Tx result on other nodes. The node should be upgraded and do a bootstrap. Hot upgrade without a correct bootstrap can lead to inconsistency between nodes.
 * it fails on all node and the user has to change the Tx and retry. It's a normal case that don't lead to inconsistency between nodes.
As we didn't define the logic of Tx execution depending on the error, I choose to end the Tx execution.
It avoids having a Tx that always restart even after a reboot.